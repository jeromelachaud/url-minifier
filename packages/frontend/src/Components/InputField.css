.input-field {
  border:1px solid lightgrey;
  -webkit-appearance:textfield;
  width: 100%;
  padding: 0 8px 0;
  line-height: 40px;
  outline: none;
}

.input-field:focus {
  border-color: grey;
}
