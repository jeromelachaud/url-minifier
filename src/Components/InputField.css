.input-field {
  border: 1px solid lightgrey;
  -webkit-appearance: textfield;
  width: 100%;
  padding: 0 10px 0;
  line-height: 60px;
  outline: none;
}

.input-field:focus {
  border-color: grey;
}
