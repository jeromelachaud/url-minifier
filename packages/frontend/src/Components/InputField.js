import React from 'react'
import { Label } from './Label'
import PropTypes from 'prop-types'
import './InputField.css'

export const InputField = props => (
  <div>
    <Label
      htmlFor="InputField"
      text="Enter the URL to minify"/>

    <input
      required
      size="80"
      type="url"
      aria-label="Enter the URL to minify"
      id="InputField"
      className="input-field"
      placeholder="URL"
      onChange={props.onChange}>
    </input>
  </div>

)

InputField.propTypes = {
  onChange: PropTypes.func,
}
