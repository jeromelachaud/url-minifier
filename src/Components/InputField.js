import PropTypes from 'prop-types'
import React from 'react'
import './InputField.css'
import { Label } from './Label'

export const InputField = props => (
  <div>
    <Label id={props.id} htmlFor={props.htmlFor} text={props.labelText} />

    <input
      required
      size="80"
      type={props.type}
      aria-label={props.ariaLabel}
      id={props.id}
      className="input-field"
      placeholder={props.placeholder}
      onChange={props.onChange}
    ></input>
  </div>
)

InputField.propTypes = {
  onChange: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired,
  ariaLabel: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  htmlFor: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  labelText: PropTypes.string.isRequired,
}
