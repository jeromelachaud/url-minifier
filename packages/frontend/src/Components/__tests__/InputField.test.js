import React from 'react'
import { shallow } from 'enzyme'
import { InputField } from '../InputField'

const onChange = function () { }
const type = 'string'
const ariaLabel = 'string'
const placeholder = 'string'
const htmlFor = 'string'
const id = 'string'
const labelText = 'string'
const text = 'string'

it('renders correctly', () => {
  const tree = shallow(
    <InputField
      onChange={onChange}
      type={type}
      ariaLabel={ariaLabel}
      placeholder={placeholder}
      htmlFor={htmlFor}
      id={id}
      labelText={labelText}
      text={text}
    />
  )
  expect(tree).toMatchSnapshot()
})
