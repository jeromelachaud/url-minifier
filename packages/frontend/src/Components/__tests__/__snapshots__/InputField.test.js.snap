// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <Component
    htmlFor="string"
    text="string"
  />
  <input
    aria-label="string"
    className="input-field"
    id="string"
    onChange={[Function]}
    placeholder="string"
    required={true}
    size="80"
    type="string"
  />
</div>
`;
